examples/CloudFirestore.md

```diff
@@ -0,0 +1,29 @@
+# Using DataLoader with Cloud Firestore
+
+Cloud Firestore is a "NoSQL" document-oriented database which supports [in operator](https://firebase.google.com/docs/firestore/query-data/queries#in_not-in_and_array-contains-any),
+making it can be used with DataLoader.
+
+Here we build an example Cloud Firestore DataLoader using [firebase-admin](https://firebase.google.com/docs/admin/setup?hl=en).
+
+```js
+const admin = require("firebase-admin");
+
+admin.initializeApp({
+  // You need to Initialize the SDK
+});
+
+const datastoreLoader = new DataLoader(
+  async (keys) => {
+    const snapshot = await admin
+      .firestore()
+      .collection(`users`)
+      .where(admin.firestore.FieldPath.documentId(), `in`, keys)
+      .get();
+    return snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
```

Comment:
Does firestore guarantee the results will be in the same order as the `keys` input? If you're not sure this is the case, I think it might be safer if this line was something like:

```suggestion
    return keys.map(key => {
      const doc = snapshot.docs.find(doc => doc.id === key);
      return doc ? { ...doc.data(), id: doc.id } : null;
    });
```

(Obviously test this works before adopting it; I've never used Firestore.)

Reply:
@benjie 
Thank you for your review.

As you said, firestore is retrieved in ascending order by document ID by default, so I think sorting by input key is necessary.

I have added the correction and comments.

I have also implemented and confirmed the test in [my repository](https://github.com/shoNagai/firestore-dataloader/blob/main/src/__tests__/firestoreDataLoader.test.ts).